diff -r 4532fe5ee826808805b373e9296d2baf4db254d9 -r 1060ea1c96dd9dfb031bd1987ceb2bf437a096a3 testing/path/test_svnauth.py
--- a/testing/path/test_svnauth.py	Fri Oct 19 11:00:30 2012 +0200
+++ b/testing/path/test_svnauth.py	Sun Oct 21 20:37:54 2012 +0200
@@ -261,7 +261,10 @@
         u.propget('foo')
         assert '--username="foo" --password="bar"' in u.commands[0]
 
-class pytest_funcarg__setup:
+def pytest_funcarg__setup(request):
+    return Setup(request)
+
+class Setup:
     def __init__(self, request):
         if not svnbin:
             py.test.skip("svn binary required")
